spec/unit/driver/auto_scaling_group_spec.rb

```diff
@@ -27,83 +27,152 @@
       @driver.warm_up 1
     end
 
     it 'should ignore warming up if desired number is less than min size of the group' do
       expect(group).not_to receive(:set_desired_capacity)
-      expect(group).not_to receive(:desired_capacity)
+      expect(@driver).not_to receive(:wait_for_desired_capacity)
       @driver.warm_up 0
     end
 
     it 'should warm up to maximum if desired number is greater than maximum size of group' do
-      expect(group).to receive(:auto_scaling_instances){[instance1, instance2, instance3, instance4]}
       expect(group).to receive(:set_desired_capacity).with(4)
-      expect(group).to receive(:desired_capacity).and_return(4)
+      expect(@driver).to receive(:wait_for_desired_capacity)
       @driver.warm_up 5
     end
   end
 
-  describe '#healthy_instance_count' do
-    it 'should respond with the number of instances that are HEALTHY' do
-      instance5 =  double('instance1', :health_status => 'UNHEALTHY')
-      allow(group).to receive(:auto_scaling_instances){[instance1, instance2, instance3, instance4, instance5]}
-      expect(@driver.send(:healthy_instance_count)).to eql 4
+  describe '#healthy_instance_ids' do
+    it 'returns the ids of all instances that are healthy' do
+      instance1 =  double('instance1', :health_status => 'HEALTHY', id: 'instance1')
+      instance2 =  double('instance2', :health_status => 'HEALTHY', id: 'instance2')
+      instance3 =  double('instance3', :health_status => 'UNHEALTHY', id: 'instance3')
+      instance4 =  double('instance4', :health_status => 'HEALTHY', id: 'instance4')
+      allow(group).to receive(:auto_scaling_instances){[instance1, instance2, instance3, instance4]}
+
+      expect(@driver.healthy_instance_ids).to eql ['instance1', 'instance2', 'instance4']
     end
 
-    it 'health check should be resilient against intermittent errors' do
-      instance5 = double('instance5')
-      expect(instance5).to receive(:health_status).and_raise(StandardError)
-      allow(group).to receive(:auto_scaling_instances){ [ instance5 ] }
-      expect(@driver.send(:healthy_instance_count)).to eql -1
+    it 'returns the ids of all instances that are healthy (case insensitive)' do
+      instance1 =  double('instance1', :health_status => 'HealThy', id: 'instance1')
+      allow(group).to receive(:auto_scaling_instances){[instance1]}
+
+      expect(@driver.healthy_instance_ids).to eql ['instance1']
     end
+  end
 
-    context 'when an elb is associated with the auto scaling group' do
-      it 'should not include instances that are HEALTHY but not associated with the elb' do
-        instance_collection = double('instance_collection', :health => [{:instance => ec2_instance1, :state => 'InService'}])
-        load_balancer = double('load_balancer', :instances => instance_collection)
-        allow(group).to receive(:load_balancers) { [load_balancer] }
-        allow(group).to receive(:auto_scaling_instances) { [instance1, instance2] }
```

Comment:
This is the existing test case for the use case of instances from different ASG's attached to the ELB (which would be common in a blue/green scenario).